client/plugin-src/cordova-plugin-ssh/src/android/Ssh.java

```diff
@@ -0,0 +1,151 @@
+  JSch jsch;
+  ArrayList<Session> sessions;
+
+  @Override
+  protected void pluginInitialize() {
+    jsch = new JSch();
+    sessions = new ArrayList<Session>();
+  }
+
+  @Override
+  public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+    try {
+      if (action.equals("connectPassword")) {
+        String host = args.getString(0);
+        int port = args.getInt(1);
+        String username = args.getString(2);
+        String password = args.getString(3);
+
+        this.connect(host, port, username, password, callbackContext);
+        return true;
+      } else if (action.equals("connectKey")) {
+        String host = args.getString(0);
+        int port = args.getInt(1);
+        String username = args.getString(2);
+        String privateKey = args.getString(3);
+
+        this.connectKey(host, port, username, privateKey, callbackContext);
+        return true;
+      } else if (action.equals("sendCommand")) {
+        int connectionId = args.getInt(0);
+        String command = args.getString(1);
+
+        this.sendCommand(connectionId, command, callbackContext);
+        return true;
+      } else if (action.equals("disconnect")) {
+        int connectionId = args.getInt(0);
+        this.disconnect(connectionId, callbackContext);
+        return true;
+      }
+    } catch (Exception e) {
+      callbackContext.error(e.toString());
+    }
+    return false;
+  }
+
+  private Session getSession(int idConnection) throws IndexOutOfBoundsException {
+    return sessions.get(idConnection);
+  }
+
+  private void connect(String host, int port, String username, String password, CallbackContext callbackContext) throws JSONException, JSchException {
+    int idConnection = sessions.size();
+    Session session = jsch.getSession(username, host, port);
+    sessions.add(session);
+
+    if (password != null) {
+      session.setPassword(password);
+    }
+
+    // Avoid asking for key confirmation
+    Properties prop = new Properties();
+    prop.put("StrictHostKeyChecking", "no"); //TODO this is bad
+    session.setConfig(prop);
+
+    session.connect();
+
+    callbackContext.success(idConnection);
+  }
+
+  private void connectKey(String host, int port, String username, String privateKey, CallbackContext callbackContext) throws JSONException, JSchException {
+    // posts on StackOverflow indicate that this is needed
+    jsch.removeAllIdentity();
+    jsch.addIdentity("identity", privateKey.getBytes(), null, null);
+
+    this.connect(host, port, username, null, callbackContext);
+  }
+
+  private void sendCommand(int idConnection, String command, CallbackContext callbackContext) throws JSchException, IOException, JSONException {
+    Session session = this.getSession(idConnection);
+    ChannelExec channel = (ChannelExec)session.openChannel("exec");
+    channel.setCommand(command);
+
+    ByteArrayOutputStream output = new ByteArrayOutputStream();
```

Comment:
http://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html

You can just write to the bytearrayoutputstream and then call toByteArray at the end to retrieve the full thing